From 7776e413cc6ca5f1f9b7c9697806d3b98d7ed9b7 Mon Sep 17 00:00:00 2001
From: Rob Clark <rclark@redhat.com>
Date: Wed, 17 Dec 2014 21:38:54 -0500
Subject: [drm] radeon: add connector quirk for fujitsu board

Message-id: <1418852380-13061-44-git-send-email-rclark@redhat.com>
Patchwork-id: 102176
O-Subject: [RHEL7 drm 43/89] drm/radeon: add connector quirk for fujitsu board
Bugzilla: 1173317
RH-Acked-by: Jerome Glisse <jglisse@redhat.com>
RH-Acked-by: Dave Airlie <airlied@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1173317
Upstream: since v3.16.4

commit d6ba8818372137f7cabcbf36719eff2138c4addb

Author:     Alex Deucher <alexander.deucher@amd.com>
AuthorDate: Mon Sep 8 13:55:51 2014 -0400
Commit:     Rob Clark <rclark@redhat.com>
CommitDate: Wed Dec 10 17:19:55 2014 -0500

    drm/radeon: add connector quirk for fujitsu board

    commit 1952f24d0fa6292d65f886887af87ba8ac79b3ba upstream.

    Vbios connector table lists non-existent VGA port.

    Bug:
    https://bugs.freedesktop.org/show_bug.cgi?id=83184

    Signed-off-by: Alex Deucher <alexander.deucher@amd.com>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

Signed-off-by: Rob Clark <rclark@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/gpu/drm/radeon/radeon_atombios.c b/drivers/gpu/drm/radeon/radeon_atombios.c
index 4ad00b1..be6705e 100644
--- a/drivers/gpu/drm/radeon/radeon_atombios.c
+++ b/drivers/gpu/drm/radeon/radeon_atombios.c
@@ -447,6 +447,13 @@ static bool radeon_atom_apply_quirks(struct drm_device *dev,
   }
  }
 
+ /* Fujitsu D3003-S2 board lists DVI-I as DVI-I and VGA */
+ if ((dev->pdev->device == 0x9805) &&
+     (dev->pdev->subsystem_vendor == 0x1734) &&
+     (dev->pdev->subsystem_device == 0x11bd)) {
+  if (*connector_type == DRM_MODE_CONNECTOR_VGA)
+   return false;
+ }
 
  return true;
 }
-- 
1.7.1